src/core/mac/mac.cpp

```diff
@@ -653,45 +661,58 @@ void Mac::PerformOperation(void)
     // operations since radio should remain in receive mode after
```

Comment:
Suggestion: Instead of adding a `if (mEnabled)` which requires the a lot of the code to shifted to right (to be inside the `if`) how about if we handle the "not enabled" case at the top of the method and do `ExitNow()`. It makes the diff smaller. Something like this:

```
if (!mEnabled) 
{
        mPendingWaitingForData   = false;
        mPendingTransmitOobFrame = false;
        mPendingActiveScan       = false;
        mPendingEnergyScan       = false;
        mPendingTransmitBeacon   = false;
        mPendingTransmitData     = false;
        mOobFrame                = NULL;
        ExitNow();
}
````

This way we also skip the log (which adds "Starting operation "<something>")  which is at the end of method when `mEnabled` is false.